src/index.js

```diff
@@ -0,0 +1,167 @@
+    if (article.description) {
+      const divNode = document.createElement("div");
+
+	  //Two divs are generated due to the final layout for wide screens
+    // one with img and one without;
+    const articleSource= article.source.id ? article.source.id : article.source.name
+
+      if (article.urlToImage) {
+        divNode.className = "article-div-w-img";
+
+        divNode.innerHTML = `<div class="title"><h2>${article.title}</h2></div>
+
+                    <div  class="date"><h4>${date}</h4></div>
+
+                    <div class="img"><img src="${article.urlToImage}"></div>
+
+                    <div class="content"><p>${article.description}</p></div>
+
+                    <div  class="source"><h4>Source: ${
+                      article.source.name
+                    }</h4></div>
+
+                    <div class="more-news"><button class="news" data-source="${articleSource}">More headlines from this source</button></div>
+
+                    <div class="link"><a href="${
+                      article.url
+                    }">Read all about it</a></div>`;
+      } else {
+        divNode.className = "article-div";
+
+        divNode.innerHTML = `<div class="title"><h2>${article.title}</h2></div>
+
+                    <div class="date"><h4>${date}</h4></div>
+
+                    <div class="content"><p>${article.description}</p></div>
+
+                    <div  class="source"><h4>Source: ${article.source.name}</h4></div>
+
+                    <div class="more-news"><button class="news" data-source="${articleSource}>More headlines from this source</button></div>
+
+                    <div class="link"><a href="${
+                      article.url
+                    }">Read all about it</a></div>`;
+      }
+
+	  //   The newly created article div is added to the newsfeed
+      const parentNode = document.querySelector("#newsfeed");
+      parentNode.appendChild(divNode);
+    };
+  
+  });
+}
+
+
+//Add event listener to the search form
+document.querySelector("form").addEventListener("submit", e => {
+  e.preventDefault();
+  let searchItem = document.querySelector("#search").value;
+  //The exisiting newsfeed is cleared before search results displayed
+  document.querySelector("#newsfeed").textContent = "";
+
+  const searchURL =
+    "https://newsapi.org/v2/everything?" +
+    `q=${searchItem}&` +
+    "from=2018-09-14&" +
+    "sortBy=popularity&" +
+    "language=en&" +
+    "apiKey=7318e7fb7dc04d14af2f0fd675cfda53";
+
+  const reqURL = new Request(searchURL);
+
+  fetch(reqURL)
+    .then(function(response) {
+      return response.json();
+    })
+
+    .then(function(body) {
+      displayArticles(body);
+    });
+
+  document.querySelector("#search").value = "";
+});
+
+	// Tried adding an event listener to 'More news from this source button', but it does not seem to work.
+  document.querySelectorAll(".news").addEventListener("click", e => {
```

Comment:
querySelectorAll returns a NodeList and we can't set an event handler on it. It would have to be done on each element in list individually